### ISSUE TITLE
webapp custom generator should not require to set a CMD configuration

### ISSUE THREAD

**Issue by sunix** [eclipse/jkube#1284]
### Description
Setting up a webapp (war) project with this configuration in the `build.gradle` file
```
kubernetes {
    generator {
        config {
            'webapp' {
                from = "tomcat:jdk11-openjdk-slim"
                targetDir = "/usr/local/tomcat/webapps"
            }
        }
    }
}
```
is failling to build with the error
```
> Task :k8sBuild FAILED
k8s: Running in Kubernetes mode
k8s: Running generator webapp
k8s: webapp: Using tomcat:jdk11-openjdk-slim as base image for webapp

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':k8sBuild'.
> Argument conflict: either shell or args should be specified and only in one form.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':k8sBuild'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:188)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:186)
[...]
Caused by: java.lang.IllegalArgumentException: Argument conflict: either shell or args should be specified and only in one form.
        at org.eclipse.jkube.kit.config.image.build.Arguments.validate(Arguments.java:109)
        at org.eclipse.jkube.kit.config.image.build.BuildConfiguration.initAndValidate(BuildConfiguration.java:336)
        at org.eclipse.jkube.kit.build.service.docker.helper.ConfigHelper.initAndValidate(ConfigHelper.java:245)
        at org.eclipse.jkube.kit.build.service.docker.helper.ConfigHelper.initAndValidate(ConfigHelper.java:134)
        at org.eclipse.jkube.kit.build.service.docker.helper.ConfigHelper.initImageConfiguration(ConfigHelper.java:209)
[...]
BUILD FAILED in 814ms
1 actionable task: 1 executed
```

Adding `cmd` to the configuration is fixing it:
```
kubernetes {
    generator {
        config {
            'webapp' {
                from = "tomcat:jdk11-openjdk-slim"
                targetDir = "/usr/local/tomcat/webapps"
                cmd = "catalina.sh run"
            }
        }
    }
}
```

However, it should not fail without `cmd` and use the command that is provided by the container image.

### Info

* Eclipse JKube version : 1.6.0

* Kubernetes / Red Hat OpenShift setup and version :
Minikube

* Sample Reproducer Project
Using the `webapp` quickstart and adding the configuration to it.

**Comment by sunix** [comment]
also the message is misleading:
- it sounds like there is a conflict but there isn't
- what is `shell`?
- `in one form`? maybe rephrase this

**Comment by manusa** [comment]
I'm not sure why this is causing trouble here and not in other kinds of generators such as JavaExec, where there is no definition for these fields.

**Comment by manusa** [comment]
OK, the problems seems to be here:
https://github.com/eclipse/jkube/blob/f3f44a12d96e9b65ddfd1568ea63bf521ba8a174/jkube-kit/generator/webapp/src/main/java/org/eclipse/jkube/generator/webapp/WebAppGenerator.java#L111
We probably need to include the `.cmd` call only if `getDockerRunCommand(handler)` is not null.
